File: UVa 10083 division/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main {

    public static void main(String[] args) throws IOException {
        InputStreamReader isr = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(isr);
        StringBuffer sb = new StringBuffer("");
        String m = "";
        while ((m = br.readLine()) != null) {
            String[] str = m.split(" ");
            int t = Integer.parseInt(str[0]);
            int a = Integer.parseInt(str[1]);
            int b = Integer.parseInt(str[2]);

            //to check if the the result is the base is not equal to 0.
            if (t == 1) {
                sb.append("(").append(t).append("^").append(a).append("-1)/(").append(t).append("^").append(b).append("-1) ").append("is not an integer with less than 100 digits.\n");
                continue;
            }


            if (a == b) {
                sb.append("(").append(t).append("^").append(a).append("-1)/(").append(t).append("^").append(b).append("-1) ").append("1\n");
                continue;
            }

            //to check if the the result is an integer
            if (a % b != 0 || a < b) {
                sb.append("(").append(t).append("^").append(a).append("-1)/(").append(t).append("^").append(b).append("-1) ").append("is not an integer with less than 100 digits.\n");
                continue;
            }


            //to check if the the result is less than 100 digits
            if ( (a - b) * Math.log10(t) > 99) {
                sb.append("(").append(t).append("^").append(a).append("-1)/(").append(t).append("^").append(b).append("-1) ").append("is not an integer with less than 100 digits.\n");
                continue;
            }


            BigInteger bg = (BigInteger.valueOf(t).pow(a).subtract(BigInteger.ONE)).divide(BigInteger.valueOf(t).pow(b).subtract(BigInteger.ONE));

            sb.append("(").append(t).append("^").append(a).append("-1)/(").append(t).append("^").append(b).append("-1) ").append(bg).append("\n");
        }
        System.out.print(sb);
    }
}

File: UVa 10083 division/main.py
import sys
import math

sys.stdin = open('input.txt')
while True:
    try:
        t, a, b = map(int, raw_input().split())
    except:
        break
    print "(%d^%d-1)/(%d^%d-1)" % (t, a, t, b),
    if t == 1:
        print("is not an integer with less than 100 digits.")
        continue
    if a == b:
        print 1
        continue
    if a % b != 0 or a < b:
        print("is not an integer with less than 100 digits.")
        continue
    if (a - b) * math.log10(t) > 99:
        print("is not an integer with less than 100 digits.")
        continue
    X = pow(t, a) - 1
    Y = pow(t, b) - 1
    if Y == 0:
        print("is not an integer with less than 100 digits.")
        continue
    tmp = X % Y
    X = X / Y
    print(X)
